src/Mod/PartDesign/App/FeaturePad.cpp

```diff
@@ -66,14 +65,18 @@ Pad::Pad()
 {
     addSubType = FeatureAddSub::Additive;
 
-    ADD_PROPERTY_TYPE(Type,(0L),"Pad",App::Prop_None,"Pad type");
+    ADD_PROPERTY_TYPE(Type, (0L), "Pad", App::Prop_None, "Pad type");
     Type.setEnums(TypeEnums);
-    ADD_PROPERTY_TYPE(Length,(100.0),"Pad",App::Prop_None,"Pad length");
-    ADD_PROPERTY_TYPE(Length2,(100.0),"Pad",App::Prop_None,"Second Pad length");
-    ADD_PROPERTY_TYPE(UpToFace,(0),"Pad",App::Prop_None,"Face where pad will end");
-    ADD_PROPERTY_TYPE(Offset,(0.0),"Pad",App::Prop_None,"Offset from face in which pad will end");
+    ADD_PROPERTY_TYPE(Length, (100.0), "Pad", App::Prop_None,"Pad length");
+    ADD_PROPERTY_TYPE(Length2, (100.0), "Pad", App::Prop_None,"Second Pad length");
+    ADD_PROPERTY_TYPE(UseCustomVector, (0), "Pad", App::Prop_None, "Use custom vector for pad direction");
+    ADD_PROPERTY_TYPE(XSkew, (1.0), "Pad", App::Prop_None, "x-component of pad direction vector");
+    ADD_PROPERTY_TYPE(YSkew, (1.0), "Pad", App::Prop_None, "y-component of pad direction vector");
+    ADD_PROPERTY_TYPE(ZSkew, (1.0), "Pad", App::Prop_None, "z-component of pad direction vector");

```

Comment:
Please convert to PropertyVector. This will enable vector assignment in expressions.